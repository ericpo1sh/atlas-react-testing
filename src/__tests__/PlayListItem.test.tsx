import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import PlayListItem from "../components/PlayListItem";

test("Play List Item Renders Correctly With valid props", () => {
  const { container } = render(<PlayListItem title='test' artist='testerEric' duration='7:77' className=""/>);
  expect(container).toMatchSnapshot();
});

test("Play List Item Renders Correctly With valid props v2", () => {
  const { container } = render(<PlayListItem title='test-2' artist='testerEric-2' duration='8:88' className="bg-slate-200 rounded"/>);
  expect(container).toMatchSnapshot();
});

test("Play List Item Renders Correctly Without Props", () => {
  const { container } = render(<PlayListItem/>);
  expect(container).toMatchSnapshot();
})
