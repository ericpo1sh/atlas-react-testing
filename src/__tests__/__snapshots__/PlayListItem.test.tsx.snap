// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cover Art Renders Correctly 1`] = `
<div>
  <div
    class="w-96 flex-col mt-2.5"
  >
    <div
      class="flex justify-between undefined"
    >
      <div
        class="text-left"
      >
        <p
          class="text-base font-bold"
        >
          test
        </p>
        <p
          class="text-jacarta-600 font-medium"
        >
          testerEric
        </p>
      </div>
      <p
        class="text-jacarta-400 font-medium text-lg mt-2"
      >
        7:77
      </p>
    </div>
  </div>
</div>
`;

exports[`Play List Item Renders Correctly 1`] = `
<div>
  <div
    class="w-96 flex-col mt-2.5"
  >
    <div
      class="flex justify-between "
    >
      <div
        class="text-left"
      >
        <p
          class="text-base font-bold"
        >
          test
        </p>
        <p
          class="text-jacarta-600 font-medium"
        >
          testerEric
        </p>
      </div>
      <p
        class="text-jacarta-400 font-medium text-lg mt-2"
      >
        7:77
      </p>
    </div>
  </div>
</div>
`;

exports[`Play List Item Renders Correctly With valid props 1`] = `
<div>
  <div
    class="w-96 flex-col mt-2.5"
  >
    <div
      class="flex justify-between "
    >
      <div
        class="text-left"
      >
        <p
          class="text-base font-bold"
        >
          test
        </p>
        <p
          class="text-jacarta-600 font-medium"
        >
          testerEric
        </p>
      </div>
      <p
        class="text-jacarta-400 font-medium text-lg mt-2"
      >
        7:77
      </p>
    </div>
  </div>
</div>
`;

exports[`Play List Item Renders Correctly With valid props v2 1`] = `
<div>
  <div
    class="w-96 flex-col mt-2.5"
  >
    <div
      class="flex justify-between bg-slate-200 rounded"
    >
      <div
        class="text-left"
      >
        <p
          class="text-base font-bold"
        >
          test-2
        </p>
        <p
          class="text-jacarta-600 font-medium"
        >
          testerEric-2
        </p>
      </div>
      <p
        class="text-jacarta-400 font-medium text-lg mt-2"
      >
        8:88
      </p>
    </div>
  </div>
</div>
`;

exports[`Play List Item Renders Correctly Without Props 1`] = `
<div>
  <div
    class="w-96 flex-col mt-2.5"
  >
    <div
      class="flex justify-between undefined"
    >
      <div
        class="text-left"
      >
        <p
          class="text-base font-bold"
        />
        <p
          class="text-jacarta-600 font-medium"
        />
      </div>
      <p
        class="text-jacarta-400 font-medium text-lg mt-2"
      />
    </div>
  </div>
</div>
`;
